content/src/ports/postgres.ts

```diff
@@ -87,7 +87,12 @@ export async function createDatabaseComponent(
   }
 
   async function* streamQuery<T>(sql: SQLStatement, config?: { batchSize?: number }): AsyncGenerator<T> {
-    const client = await pool.connect()
+    const client = new Client({
+      ...finalOptions,
+      query_timeout: components.env.getConfig<number>(EnvironmentConfig.PG_STREAM_QUERY_TIMEOUT)
```

Comment:
can we move `components.env.getConfig<number>(EnvironmentConfig.PG_STREAM_QUERY_TIMEOUT)` to a variable defined in the constructor? Just to have all the env configs together